app/controllers/application_controller.rb

```diff
@@ -1,10 +1,15 @@
 class ApplicationController < ActionController::Base
   before_filter :configure_permitted_parameters, if: :devise_controller?
+  before_filter :set_current_user
   before_action :set_locale
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery
 
+  def set_current_user
+    TestSetting.current_user = current_user
```

Comment:
not so good idea set to TestSetting current_user after each request. Why don't use device method `current_user` and use it only on needed places.